src/components/shared/UserAvatar/UserAvatar.tsx

```diff
@@ -59,71 +56,71 @@ const displayName = 'UserAvatar';
 const UserAvatar = ({
   address,
   avatarURL,
+  banned = false,
   className,
-  showInfo,
-  showLink,
   notSet,
   popperOptions,
+  showInfo,
+  showLink,
   size,
-  banned = false,
   user,
-  ...rest
+  ...reputationProps
 }: Props) => {
-  // const username = getUsername(user);
-  // const username = 'tempUser';
-  // let popoverProps: InfoPopoverProps = {
-  //   popperOptions,
-  //   trigger: showInfo ? 'click' : 'disabled',
-  //   user,
-  //   showArrow: popperOptions && popperOptions.showArrow,
-  // };
-  // if ('colony' in rest) {
-  //   const { colony, domainId } = rest;
-  //   popoverProps = {
-  //     ...popoverProps,
-  //     banned,
-  //     colony,
-  //     domainId,
-  //   };
-  // }
-  // const avatar = (
-  //   <InfoPopover {...popoverProps}>
-  //     <div
-  //       // @ts-ignore
-  //       className={getMainClasses({}, styles, {
-  //         showOnClick: popoverProps.trigger === 'click',
-  //       })}
-  //     >
-  //       <Avatar
-  //         avatarURL={avatarURL}
-  //         className={className}
-  //         notSet={typeof notSet === 'undefined' ? true : notSet}
-  //         placeholderIcon="circle-person"
-  //         seed={address && address.toLowerCase()}
-  //         size={size}
-  //         title={showInfo ? '' : username || address}
-  //       />
-  //     </div>
-  //   </InfoPopover>
-  // );
-  // if (showLink && username) {
-  //   // Won't this always be lowercase?
-  //   return <Link to={`/user/${username.toLowerCase()}`}>{avatar}</Link>;
-  // }
-  // return avatar;
-  return (
-    <Avatar
-      avatarURL={avatarURL}
-      className={className}
-      notSet={typeof notSet === 'undefined' ? true : notSet}
-      placeholderIcon="circle-person"
-      seed={address && address.toLowerCase()}
-      size={size}
-      title={
-        showInfo ? '' : user?.profile?.displayName || user?.name || address
+  const username = getUsername(user);
+  const trigger = showInfo ? 'click' : 'disabled';
+  const showArrow = popperOptions && popperOptions.showArrow;
+
+  const renderContent = useMemo(() => {
+    if ('colony' in reputationProps) {
+      const { colony } = reputationProps;
+      if (typeof colony !== 'undefined') {
+        return (
+          <MemberInfoPopover
```

Comment:
Unless there's something super blocking, `MemberInfoPopover` and `UserInfoPopover` should be just one component